### PR TITLE
Implement Third Page of Core Demo

### PR DESCRIPTION
**Pull request by MSFTFox**
Implement third page of core demo. This page shows off various blend modes with images.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc-samples/32)

### CODE REVIEW

File: Scenarios/CGCT/CGCT/DemoScenarioImages.m [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    int blendModeIndex = 0;
+    for (CGFloat x = 0; x < 1; x += .34) {
+        for (CGFloat y = 0; y < 1; y += .25) {
+            CGContextSetRGBFillColor(context,
+                                     fillColors[blendModeIndex * 4],
+                                     fillColors[blendModeIndex * 4 + 1],
+                                     fillColors[blendModeIndex * 4 + 2],
+                                     fillColors[blendModeIndex * 4 + 3]);
+            CGRect pane = CGRectMake(x * bounds.size.width, y * bounds.size.height, catSize, catSize * aspectRatio);
```
Comment (DHowett-MSFT) [inline Scenarios/CGCT/CGCT/DemoScenarioImages.m]: `cat` :laughing:

File: Scenarios/CGCT/CGCT/DemoScenarioImages.m [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    return @"Images";
+}
+
+- (UIColor*)backgroundColor {
+    return [UIColor colorWithRed:.72 green:.6 blue:.4 alpha:1];
+}
+
+- (instancetype)init {
+    if (self = [super init]) {
+        patternImage = [UIImage imageWithContentsOfFile:[[NSBundle mainBundle] pathForResource:@"sunset" ofType:@"png"]];
```
Comment (DHowett-MSFT) [inline Scenarios/CGCT/CGCT/DemoScenarioImages.m]: nit: not a pattern, and ivars usually begin with `_`

File: Scenarios/CGCT/CGCT/DemoScenarioImages.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import "DemoScenario.h"
+
+@interface DemoScenarioImages : DemoScenario
+- (void)drawDemoIntoContext:(CGContextRef)context withFrame:(CGRect)bounds;
```
Comment (DHowett-MSFT) [inline Scenarios/CGCT/CGCT/DemoScenarioImages.h]: technically you need not specify this method

**Review by DHowett-MSFT — APPROVED** [review]
same question about PD

File: Scenarios/CGCT/CGCT/DemoScenarioImages.m [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                                     fillColors[blendModeIndex * 4 + 2],
+                                     fillColors[blendModeIndex * 4 + 3]);
+            CGRect pane = CGRectMake(x * bounds.size.width, y * bounds.size.height, catSize, catSize * aspectRatio);
+            CGContextDrawImage(context, pane, patternImage.CGImage);
+
+            CGContextSetBlendMode(context, blendModes[blendModeIndex]);
+            CGMutablePathRef blendRect = CGPathCreateMutable();
+            CGPathAddRect(blendRect, NULL, pane);
+            CGContextAddPath(context, blendRect);
+            CGContextDrawPath(context, kCGPathFill);
```
Comment (DHowett-MSFT) [inline Scenarios/CGCT/CGCT/DemoScenarioImages.m]: youuuuuu could just CGContextFillRect

File: Scenarios/CGCT/CGCT/DemoScenarioImages.m [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        for (CGFloat y = 0; y < 1; y += .25) {
+            CGContextSetRGBFillColor(context,
+                                     fillColors[blendModeIndex * 4],
+                                     fillColors[blendModeIndex * 4 + 1],
+                                     fillColors[blendModeIndex * 4 + 2],
+                                     fillColors[blendModeIndex * 4 + 3]);
+            CGRect pane = CGRectMake(x * bounds.size.width, y * bounds.size.height, catSize, catSize * aspectRatio);
+            CGContextDrawImage(context, pane, patternImage.CGImage);
+
+            CGContextSetBlendMode(context, blendModes[blendModeIndex]);
```
Comment (DHowett-MSFT) [inline Scenarios/CGCT/CGCT/DemoScenarioImages.m]: also, the GState is never being saved, Each successive image is being drawn in the previous color plate's blend mode ...

**Review by DHowett-MSFT — CHANGES_REQUESTED** [review]

File: Scenarios/CGCT/CGCT/DemoScenarioImages.m [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    CGContextScaleCTM(context, 1.0f, -1.0f);
+    CGContextTranslateCTM(context, bounds.size.width * .03, -bounds.size.height * .98);
+    CGContextSaveGState(context);
+
+    CGFloat aspectRatio = patternImage.size.height / patternImage.size.width;
+    CGFloat catSize = bounds.size.width / 3.75;
+    CGContextSetRGBFillColor(context, 1, 0, 0, .3);
+    int colorIndex = 0;
+
+    CGFloat fillColors[48];
```
Comment (rajsesh) [inline Scenarios/CGCT/CGCT/DemoScenarioImages.m]: 2x2x2x2x4 = 64.

**Review by rajsesh — CHANGES_REQUESTED** [review]

**Review by DHowett-MSFT — APPROVED** [review]

**Review by rajsesh — APPROVED** [review]

**Pull request merged**